### PR TITLE
Add the DLA models

### PR DESCRIPTION
**Pull request by jm12138**
# 添加 DLA 模型系列
## DLA
* Paper：[Deep Layer Aggregation](https://arxiv.org/abs/1707.06484)
* Origin Repo：[ucbdrive/dla](https://github.com/ucbdrive/dla)
* Evaluate Transforms：

    ```python
    # backend: pil
    # input_size: 224x224
    transforms = T.Compose([
        T.Resize(256, interpolation='bilinear'),
        T.CenterCrop(224),
        T.ToTensor(),
        T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
    ])
    ```

* Model Details：

  |         Model         | Params (M) | FLOPs (G) | Top-1 (%) | Top-5 (%) |   Pretrained Model    |
  |:---------------------:|:----------:|:---------:|:---------:|:---------:|:---------------------:|
  | DLA-34                 | 15.8       | 3.1       | 76.39     |   93.15   | [Download][dla_34]    |
  | DLA-46-c              | 1.3        | 0.5       | 64.88     |   86.29   | [Download][dla_46_c]  |
  | DLA-46x-c            | 1.1        | 0.5       | 65.98     |   86.98   | [Download][dla_46x_c] |
  | DLA-60               | 22.0       | 4.2       | 77.02     |   93.31   | [Download][dla_60]    |
  | DLA-60x             | 17.4       | 3.5       | 78.24     |   94.02   | [Download][dla_60x]   |
  | DLA-60x-c              | 1.3        | 0.6       | 67.91     |   88.43   | [Download][dla_60x_c] |
  | DLA-102                | 33.3       | 7.2       | 79.44     |   94.76   | [Download][dla_102]   |
  | DLA-102x             | 26.4       | 5.9       | 78.51     |   94.23   | [Download][dla_102x]  |
  | DLA-102x2              | 41.4       | 9.3       | 79.45     |   94.64   | [Download][dla_102x2] |
  | DLA-169                | 53.5       | 11.6      | 78.71     |   94.34   | [Download][dla_169]   |
  
  |         Model         | Params (M) | FLOPs (G) | Top-1 (%) | Top-5 (%) |   Pretrained Model    |
  |:---------------------:|:----------:|:---------:|:---------:|:---------:|:---------------------:|
  | DLA34                 | 15.8       | 3.1       | 76.03     |   92.98   | [Download][dla_34]    |
  | DLA46_c              | 1.3        | 0.5       | 63.21     |   85.30   | [Download][dla_46_c]  |
  | DLA46x_c            | 1.1        | 0.5       | 64.36     |   86.01   | [Download][dla_46x_c] |
  | DLA60               | 22.0       | 4.2       | 76.10    |   92.92   | [Download][dla_60]    |
  | DLA60x             | 17.4       | 3.5       | 77.53    |   93.78   | [Download][dla_60x]   |
  | DLA60x_c              | 1.3        | 0.6       | 66.45     |   87.54   | [Download][dla_60x_c] |
  | DLA102                | 33.3       | 7.2       | 78.93     |   94.52   | [Download][dla_102]   |
  | DLA102x             | 26.4       | 5.9       | 78.10     |   94.00   | [Download][dla_102x]  |
  | DLA102x2              | 41.4       | 9.3       | 78.85     |   94.45   | [Download][dla_102x2] |
  | DLA169                | 53.5       | 11.6      | 78.09    |   94.09   | [Download][dla_169]   |

[dla_34]:https://bj.bcebos.com/v1/ai-studio-online/a4e08c790f0247c8ab44cfa9ec6264720a3fab64b51d4ee88d0e7d3511e6348a?responseContentDisposition=attachment%3B%20filename%3Ddla34%2Btricks.pdparams
[dla_46_c]:https://bj.bcebos.com/v1/ai-studio-online/245e16ae6b284b368798a6f8e3cf068e55eea96e22724ec5bff8d146c64da990?responseContentDisposition=attachment%3B%20filename%3Ddla46_c.pdparams
[dla_46x_c]:https://bj.bcebos.com/v1/ai-studio-online/b295201d245247fb8cd601b60919cabf5df51a8997d04380bd07eac71e4152dd?responseContentDisposition=attachment%3B%20filename%3Ddla46x_c.pdparams
[dla_60]:https://bj.bcebos.com/v1/ai-studio-online/e545d431a9f84bb4aecd2c75e34e6169503be2d2e8d246cb9cff393559409f7b?responseContentDisposition=attachment%3B%20filename%3Ddla60.pdparams
[dla_60x]:https://bj.bcebos.com/v1/ai-studio-online/a07ea1cec75a460ebf6dcace4ab0c8c28e923af88dd74573baaaa6db8738168d?responseContentDisposition=attachment%3B%20filename%3Ddla60x.pdparams
[dla_60x_c]:https://bj.bcebos.com/v1/ai-studio-online/0c15f589fa524d1dbe753afe2619f2fe33773c0ca6db4966a3ab8f755fca3c98?responseContentDisposition=attachment%3B%20filename%3Ddla60x_c.pdparams
[dla_102]:https://bj.bcebos.com/v1/ai-studio-online/288ca91946d04df891750eed67b3070ec38a29e9a7b24eff90c0e397d3b82c7f?responseContentDisposition=attachment%3B%20filename%3Ddla102%2Btricks.pdparams
[dla_102x]:https://bj.bcebos.com/v1/ai-studio-online/0653e6aae7594e2a8de94728f6656c375557f7960a8949a1926eb017e978c477?responseContentDisposition=attachment%3B%20filename%3Ddla102x.pdparams
[dla_102x2]:https://bj.bcebos.com/v1/ai-studio-online/80cd37d877974ad18d1ccefdae2a5c2cce1cba2831544deeaea1fa672343cc17?responseContentDisposition=attachment%3B%20filename%3Ddla102x2.pdparams
[dla_169]:https://bj.bcebos.com/v1/ai-studio-online/f299fab9020344d4aee7ccf3a79e98858494e0536bca4703a5f5152747395cca?responseContentDisposition=attachment%3B%20filename%3Ddla169.pdparams

* Citation：

    ```
    @misc{yu2019deep,
        title={Deep Layer Aggregation}, 
        author={Fisher Yu and Dequan Wang and Evan Shelhamer and Trevor Darrell},
        year={2019},
        eprint={1707.06484},
        archivePrefix={arXiv},
        primaryClass={cs.CV}
    }
    ```
## 其他
* 第一张表格的精度数据是使用 PIL 作为图像处理后端在 PaddleHapi 中跑出的结果
* 第二张表格的精度数据是使用 cv2 作为图像处理后端在 PaddleClas 中跑出的结果
* 使用 cv2 作为后端模型的精度会稍有损失
* 其中 DLA34 和 DLA102 使用的是官方后面更新的增加了 tricks 的预训练模型，会比论文中的精度高一些

### CODE REVIEW

File: configs/DLA/DLA102.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+mode: 'train'
```
Comment (littletomatodonkey) [inline configs/DLA/DLA102.yaml:1]: 你好，感谢贡献~方便添加一下dla模型的中英文说明文档吗？类似于`docs/zh_CN/models/Transformer.md`，可以提供一个`docs/zh_CN/models/DLA.md`，英文的`docs/en/models/DLA_en.md`

File: ppcls/modeling/architectures/dla.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+                 base_width=64, block=DlaBottleneck, residual_root=False,
+                 drop_rate=0.0, class_dim=1000, with_pool=True):
+        super(DLA, self).__init__()
+        self.channels = channels
+        self.class_dim = class_dim
+        self.with_pool = with_pool
+        self.cardinality = cardinality
+        self.base_width = base_width
+        self.drop_rate = drop_rate
+        assert output_stride == 32  # FIXME support dilation
```
Comment (littletomatodonkey) [inline ppcls/modeling/architectures/dla.py]: 这块目前的模型是还不支持32之外的stride吗？如果是的话，在assert后面补充下提示信息吧

File: configs/DLA/DLA102.yaml (hunk 1 above)
Comment (jm12138) [reply configs/DLA/DLA102.yaml:1]: 文档已经添加

File: ppcls/modeling/architectures/dla.py (hunk 2 above)
Comment (jm12138) [reply ppcls/modeling/architectures/dla.py]: 这个代码没有啥子意义其实，更新将其删除了

**Review by littletomatodonkey — APPROVED** [review]

**Pull request closed without merging**